File: src/Services/ProductsService/ProductsService.Domain/Product.cs
namespace ProductsService.Domain.Entities
{
    public class Product
    {
        public Guid Id { get; set; }

        public string Name { get; set; } = default!;
        public string Description { get; set; } = default!;
        public decimal Price { get; set; }
        public string Category { get; set; } = default!;
        public Boolean Status { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}

File: spa_products/src/types/product.ts
export type ProductDto = {
  id: string
  name: string
  description: string
  price: number
  category: string
  status: boolean
  createdAt: string
  updatedAt?: string
}

export type ProductFormData = {
  name: string
  description: string
  price: number
  category: string
  status?: boolean
}

export type CreateProductRequestDto = {
  name: string
  description: string
  price: number
  category: string
  status?: boolean
}

export type UpdateProductRequestDto = {
  name: string
  description: string
  price: number
  category: string
  status?: boolean
}
